File: programmers_level2/큰_수_만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string number, int k) {
    string answer = "";
    for (int i=0,idx=-1; i<number.length()-k; ++i) {
        char max = '0';
        for (int j=idx+1; j<=k+i; ++j) {
            if (max < number[j]) {
                max = number[j];
                idx = j;
            }
        }
        answer += max;
    }
    return answer;
}

File: programmers/큰 수 만들기.java
import java.util.*;
class Solution {
    public String solution(String number, int k) {
        String answer = "";
        int len=number.length()-k;
        Stack<Character> stack = new Stack<>();
        for(int i=0;i<number.length();i++){
            while(!stack.isEmpty() && stack.peek()<number.charAt(i) && k>0){
                stack.pop();
                k--;
            }
            stack.push(number.charAt(i));
        }
        for(int i=0;i<stack.size();i++){
            answer+=stack.get(i);
        }
       answer=answer.substring(0,len);
        return answer;
    }
}
